pcdsdevices/pseudopos.py

```diff
@@ -282,6 +309,314 @@ def inverse(self, real_pos):
+    # float that defines what scale to use for motors missing from scales
+    default_scale = 1
+    # bool that defines if we should warn about inconsistent axes
+    warn_inconsistent = True
+    # float that defines how much of a deviation to warn on
+    warn_deadband = 0.001
+    # bool that defines which motor to keep still when calling fix_sync
+    # if omitted, we'll keep the first real motor still
+    fix_sync_keep_still = None
+    # tuple of two floats that defines if we should add limits to the sync axis
+    sync_limits = None
+
+    def __init__(self, *args, **kwargs):
+        self._check_settings()
+        super().__init__(*args, **kwargs)
+        self._real_attrs = [attr for attr, _ in self._get_real_positioners()]
+        self.scales = self._fill_info_dict(
+            self.scales, self.default_scale, 'scales')
+        if self.sync_limits is not None:
+            self.sync._limits = tuple(self.sync_limits)
+        self._has_setup = False
+
+    def _check_settings(self):
+        """Mostly just a typo check."""
+        if self.__class__ is SyncAxis:
+            raise TypeError(
+                'SyncAxis must be subclassed with '
+                'the axes to synchronize included as '
+                'components'
+                )
+        try:
+            self.offset_mode = SyncAxisOffsetMode(self.offset_mode)
+        except ValueError:
+            try:
+                self.offset_mode = SyncAxisOffsetMode[self.offset_mode]
+            except KeyError:
+                raise ValueError(
+                    f'Invalid offset_mode: {self.offset_mode}'
+                    ) from None
+        self._check_info_dict(self.offsets, 'offsets')
+        self._check_info_dict(self.scales, 'scales')
+        if self.fix_sync_keep_still is not None:
+            try:
+                getattr(self.__class__, self.fix_sync_keep_still)
+            except AttributeError:
+                raise ValueError(
+                    'Invalid fix_sync_keep_still. Must match a motor.'
+                    ) from None
+
+    def _check_info_dict(self, setting, info_kind):
+        """Helper function to check that all keys in the dict are Cpts"""
+        if setting is not None:
+            try:
+                for attr, _ in setting.items():
+                    getattr(self.__class__, attr)
+            except AttributeError:
+                raise ValueError(
+                    f'Invalid key {attr} in {info_kind}. Must match a motor.'
+                    ) from None
+
+    def _fill_info_dict(self, setting, default, info_kind):
+        """Helper function to fill default values into the dict"""
+        if setting is None:
+            setting = {}
+        elif isinstance(setting, dict):
+            setting = copy.copy(setting)
+        else:
+            raise ValueError(
+                f'Invalid {info_kind}: {setting}, must be dict or None')
+        for attr in self._real_attrs:
+            if attr not in setting:
```

Comment:
`setting.setdefault(attr, default)` is kind of nice shorthand for this:

```
In [2]: a = {}

In [3]: a.setdefault('a', 4)
Out[3]: 4

In [4]: a
Out[4]: {'a': 4}

In [5]: a['b'] = 5

In [6]: a.setdefault('b', 7)
Out[6]: 5

In [7]: a
Out[7]: {'a': 4, 'b': 5}
```

Reply:
Never knew about this, will revise accordingly